cli/cmd/install_helm_test.go

```diff
@@ -58,6 +58,28 @@ global:
 		chartControlPlane := chartControlPlane(t, ha, additionalConfig, "333", "444")
 		testRenderHelm(t, chartControlPlane, "install_helm_output_ha_labels.golden")
 	})
+
+	t.Run("HA mode with custom namespaceSelector", func(t *testing.T) {
+		ha := true
+		additionalConfig := `
+proxyInjector:
+  namespaceSelector:
+    matchExpressions:
+    - key: config.linkerd.io/admission-webhooks
+      operator: In
+      values:
+      - enabled
+profileValidator:
+  namespaceSelector:
+    matchExpressions:
+    - key: config.linkerd.io/admission-webhooks
+      operator: In
+      values:
+      - enabled
+`
+		chartControlPlane := chartControlPlane(t, ha, additionalConfig, "111", "222")
+		testRenderHelm(t, chartControlPlane, "install_helm_output_ha_namespace_selector.golden")
```

Comment:
From looking into `install_helm_output_ha_namespace_selector.golden`, it appears those custom namespaceSelectors are not being honoured.

Reply:
good catch, @alpeb!  I think this was because this additionalConfig here was getting overwritten by the helm annotations in `testRenderHelm`.  I removed those now.